src/main/java/heronarts/glx/BGFXEngine.java

```diff
@@ -60,47 +56,66 @@ public interface Resource {
+    System.out.println("================================================");
+
     try (MemoryStack stack = MemoryStack.stackPush()) {
-      final int renderer = this.glx.flags.useOpenGL ?
-        org.lwjgl.bgfx.BGFX.BGFX_RENDERER_TYPE_OPENGL :
-        org.lwjgl.bgfx.BGFX.BGFX_RENDERER_TYPE_COUNT;
+      final int renderer =
+          this.glx.flags.useOpenGL
+              ? org.lwjgl.bgfx.BGFX.BGFX_RENDERER_TYPE_OPENGL
+              : org.lwjgl.bgfx.BGFX.BGFX_RENDERER_TYPE_COUNT;
 
       final BGFXInit init = BGFXInit.malloc(stack);
       bgfx_init_ctor(init);
-      init
-        .type(renderer)
-        .vendorId(BGFX_PCI_ID_NONE)
-        .deviceId((short) 0)
-        .resolution(res -> res
-          .width(this.glx.window.getFrameBufferWidth())
-          .height(this.glx.window.getFrameBufferHeight())
-          .reset(BGFX_RESET_VSYNC));
+      init.type(renderer)
+          .vendorId(BGFX_PCI_ID_NONE)
+          .deviceId((short) 0)
+          .debug(true)
+          .profile(true)
+          .callback(callbackInterface)
+          .resolution(
+              res ->
+                  res.width(this.glx.window.getFrameBufferWidth())
+                      .height(this.glx.window.getFrameBufferHeight())
+                      .reset(BGFX_RESET_VSYNC));
+
       switch (Platform.get()) {
         case LINUX, FREEBSD -> {
           if (glfwGetPlatform() == GLFW.GLFW_PLATFORM_WAYLAND) {
             init.platformData()
-              .ndt(GLFWNativeWayland.glfwGetWaylandDisplay())
-              .nwh(GLFWNativeWayland.glfwGetWaylandWindow(this.glx.window.handle))
-              .type(BGFX_NATIVE_WINDOW_HANDLE_TYPE_WAYLAND);
+                .ndt(GLFWNativeWayland.glfwGetWaylandDisplay())
+                .nwh(GLFWNativeWayland.glfwGetWaylandWindow(this.glx.window.handle))
+                .type(BGFX_NATIVE_WINDOW_HANDLE_TYPE_WAYLAND);
           } else {
             init.platformData()
-              .ndt(GLFWNativeX11.glfwGetX11Display())
-              .nwh(GLFWNativeX11.glfwGetX11Window(this.glx.window.handle));
+                .ndt(GLFWNativeX11.glfwGetX11Display())
+                .nwh(GLFWNativeX11.glfwGetX11Window(this.glx.window.handle));
           }
         }
-        case MACOSX -> init.platformData().nwh(GLFWNativeCocoa.glfwGetCocoaWindow(this.glx.window.handle));
-        case WINDOWS -> init.platformData().nwh(GLFWNativeWin32.glfwGetWin32Window(this.glx.window.handle));
+        case MACOSX ->
+            init.platformData().nwh(GLFWNativeCocoa.glfwGetCocoaWindow(this.glx.window.handle));
+        case WINDOWS ->
+            init.platformData().nwh(GLFWNativeWin32.glfwGetWin32Window(this.glx.window.handle));
       }
       if (!bgfx_init(init)) {
         throw new RuntimeException("Error initializing bgfx renderer");
       }
+
+      bgfx_set_debug(BGFX_DEBUG_TEXT);
+      //            bgfx_set_debug(BGFX_DEBUG_STATS | BGFX_DEBUG_TEXT);
```

Comment:
FYI, setting debug stats works, if you want an overlay on the screen with GPU/render stats, how many vertices are registered, etc, you can do `bgfx_set_debug(BGFX_DEBUG_STATS);`